skorch/callbacks/logging.py

```diff
@@ -875,3 +876,101 @@ def on_epoch_end(self, net, **kwargs):
 
         for key in filter_log_keys(epoch_logs.keys(), self.keys_ignored_):
             self.experiment.log_scalar(key + self.epoch_suffix_, epoch_logs[key], epoch)
+
+
+class MlflowLogger(Callback):
+    """
+    >>> import mlflow
+    >>> net = NeuralNetClassifier(net, callbacks=[MLflowLogger])
+    >>> with mlflow.start_run():
+    ...     net.fit(X, y)
+
+    :param run: mlflow.entities.Run
+    :param client: mlflow.tracking.MlflowClient
+    """
+    def __init__(
+        self,
+        run=None,
+        client=None,
+        create_artifact=True,
+        terminate_after_train=True,
+        log_on_batch_end=False,
+        log_on_epoch_end=True,
+        keys_ignored=None,
+        batch_suffix=None,
+        epoch_suffix=None,
+    ):
+        self.run = run
+        self.client = client
+        self.create_artifact = create_artifact
+        self.terminate_after_train = terminate_after_train
+        self.log_on_batch_end = log_on_batch_end
+        self.log_on_epoch_end = log_on_epoch_end
+        self.keys_ignored = keys_ignored
+        self.batch_suffix = batch_suffix
+        self.epoch_suffix = epoch_suffix
+
+    def initialize(self):
+        if self.run is None:
+            import mlflow
+            self.run = mlflow.active_run()
+        self.run_id = self.run.info.run_id
+        if self.client is None:
+            from mlflow.tracking import MlflowClient
+            self.client = MlflowClient()
```

Comment:
I added the support for run and client object as you ask
but by default I get the currently active experiment and a default client,
I think it's a good compromise between being customisable and being user friendly.

We can do:

```python
with mlflow.start.run():
    net = NeuralNetClassifier(..., callbacks=[MlflowLogger])
    net.fit(...)
```

instead of:

```python
client = MlflowClient
experiment = client.get_experiment_by_name(name)
run = client.create_run(experiment)
net = NeuralNetClassifier(..., callbacks=[MlflowLogger(run, client)])
net.fit(...)
```

Reply:
Yes, I guess this is okay, since the user has the choice to create the experiment and run explicitly. Could you do a minor change though and call the variables `client_` and `run_`? This is with keeping the convention to not touch the `__init__` attributes.

And if the user passes a client and run that are not `None`, still do `self.run_ = self.run` and `self.client_ = self.client`.